pkg/local_object_storage/metabase/metadata.go

```diff
@@ -0,0 +1,796 @@
+		}
+		if n == count {
+			more = true
+			if len(attrs) > 0 {
+				break
+			} // else "later" objects may have "less" ID, so we should continue
+		}
+		copy(res[insertI+1:], res[insertI:])
+		res[insertI].ID = oid.ID(id)
+		res[insertI].Attributes = collected
+		copy(collectedPrimVals[insertI+1:], collectedPrimVals[insertI:])
+		collectedPrimVals[insertI] = primDBVal
+		copy(collectedPrimKeys[insertI+1:], collectedPrimKeys[insertI:])
+		collectedPrimKeys[insertI] = primKey
+		if n < count {
+			n++
+		}
+	}
+	var newCursor *SearchCursor
+	if more {
+		newCursor = &SearchCursor{
+			Key:      collectedPrimKeys[n-1][1:],
+			ValIDOff: len(primSeekPrefix),
+		}
+	}
+	return res[:n], newCursor, nil
+}
+
+// TODO: can be merged with filtered code?
+func (db *DB) searchUnfiltered(cnr cid.ID, cursor *SearchCursor, count uint16) ([]client.SearchResultItem, *SearchCursor, error) {
+	var seekKey []byte
+	if cursor != nil {
+		seekKey = cursor.Key
+		seekKey[0] = metaPrefixID
+	} else {
+		seekKey = []byte{metaPrefixID}
+	}
+	res := make([]client.SearchResultItem, count)
+	var n uint16
+	var newCursor *SearchCursor
+	err := db.boltDB.View(func(tx *bbolt.Tx) error {
+		mb := tx.Bucket(metaBucketKey(cnr))
+		if mb == nil {
+			return nil
+		}
+
+		mbc := mb.Cursor()
+		k, _ := mbc.Seek(seekKey)
+		if cursor != nil && bytes.Equal(k, seekKey) { // cursor is the last response element, so go next
+			k, _ = mbc.Next()
+		}
+		for ; k[0] == metaPrefixID; k, _ = mbc.Next() {
+			if n == count { // there are still elements
+				newCursor = &SearchCursor{Key: res[n-1].ID[:]}
+				return nil
+			}
+			if len(k) != oid.Size+1 {
+				return invalidMetaBucketKeyErr(k, fmt.Errorf("unexpected object key len %d", len(k)))
+			}
+			res[n].ID = oid.ID(k[1:])
+			n++
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, nil, fmt.Errorf("view BoltDB: %w", err)
+	}
+	return res[:n], newCursor, nil
+}
+
+func seekKeyForAttribute(attr, fltVal string) ([]byte, []byte, error) {
+	var dbVal []byte
+	switch attr {
+	default:
+		if n, ok := new(big.Int).SetString(fltVal, 10); ok && n.Cmp(maxUint256Neg) >= 0 && n.Cmp(maxUint256) <= 0 {
+			key := make([]byte, 1+len(attr)+utf8DelimiterLen+intValLen) // prefix 1st
+			key[0] = metaPrefixAttrIDInt
+			off := 1 + copy(key[1:], attr)
+			off += copy(key[off:], utf8Delimiter)
+			prefix := key[:off]
+			putInt(key[off:off+intValLen], n)
+			return key, prefix, nil
+		}
+		dbVal = []byte(fltVal)
+	case object.FilterOwnerID, object.FilterFirstSplitObject, object.FilterParentID:
+		var err error
+		if dbVal, err = base58.Decode(fltVal); err != nil {
+			return nil, nil, fmt.Errorf("decode %q attribute value from Base58: %w", attr, err)
+		}
+	case object.FilterPayloadChecksum, object.FilterPayloadHomomorphicHash:
+		var err error
+		if dbVal, err = hex.DecodeString(fltVal); err != nil {
+			return nil, nil, fmt.Errorf("decode %q attribute value from HEX: %w", attr, err)
+		}
+	case object.FilterSplitID:
+		uid, err := uuid.Parse(fltVal)
```

Comment:
Not sure this works for `MatchCommonPrefix`.

Reply:
yeah, to support them we'd need to stringify DB values, which i didnt wanna do. So, i decided to just not support prefixing split IDs:
1. they are deprecated
2. cant imagine who can need to prefix any IDs